src/ansys/openapi/common/_base/_types.py

```diff
@@ -37,7 +37,7 @@ def to_dict(self) -> Dict[str, DeserializedType]:
     def to_str(self) -> str:
         ...
 
-    def get_real_child_model(self, data: Dict) -> str:
+    def get_real_child_model(self, data: Union[Dict, str]) -> str:
         raise NotImplementedError()
```

Comment:
Might be worth putting a note in here that this method will be overridden in the auto-generated class if the model has a discriminator.